compiler/src/utils/wasm_utils.re

```diff
@@ -460,11 +460,8 @@ module BinarySection =
           /* Now we're at the start of the section. Time to read */
           let realsize = size - (pos_in(inchan) - offset);
           let bytes = Bytes.create(realsize);
-          if (input(inchan, bytes, 0, realsize) == realsize) {
-            Some(Spec.deserialize(bytes));
-          } else {
-            process(tl);
-          };
+          really_input(inchan, bytes, 0, realsize);
```

Comment:
Is it okay that this will raise if we can't read it all?

Reply:
Indeed! If all the bytes aren't there (when the module reported that they were) means the module is corrupted.